### PR TITLE
Initial version of dark-mode

### LINKED ISSUE #13: First Issue Woo Hoo! Also its about Dark Mode
Enabling Dark Mode for Goals site 
working member - Pavan

Sample picture of Welcome site before and after enabling dark mode attached for reference

Before - 
image

After - 
image

### PR DESCRIPTION
**Pull request by Pavangbhat**
closes #13 

https://user-images.githubusercontent.com/45538392/136889236-89c0a290-3209-4058-8393-1eac433f3c20.mp4

### CODE REVIEW

**Comment by RohitKuwar** [comment]
@Pavangbhat Can we please have a border around the box/container?
Screenshot (968)

**Comment by Pavangbhat** [comment]
> > Screenshot (967)
> > @Pavangbhat Can we please fix this issue for mobile devices?
> 
> Waiting for designs 😁

**Review by Pavangbhat — COMMENTED** [review]
# Neon border :point_down: 
image

File: components/Days.vue [hunk 1]
```diff
@@ -67,11 +67,13 @@ export default {
   display: none;
 }
 .days {
-  background-color: #f8f8f8;
+  background-color: var(--background-card);
```
Comment (harshith-venkatesh) [inline components/Days.vue:70]: Any specific reason for going with `background-card` here and not `background-color` ?

File: components/Days.vue [hunk 2]
```diff
@@ -67,11 +67,13 @@ export default {
...
   display: flex;
   justify-content: space-around;
   align-items: center;
   padding: 35px 60px;
+  border: var(--border-color) solid;
```
Comment (harshith-venkatesh) [inline components/Days.vue:75]: Could we combine together, please :)
`border: 1px var(--border-color) solid;`
https://developer.mozilla.org/en-US/docs/Web/CSS/border

File: components/ThemeToggler.vue [hunk 3]
```diff
@@ -0,0 +1,69 @@
+<template>
+  <div class="theme-toggle-wrapper" @click="toggleTheme()">
+    <span>
+      <img class="theme-toggle-icon" :src="require(`~/assets/${themeImage}`)" />
+    </span>
+  </div>
+</template>
+
+<script>
+function setCookie(name, value, days) {
```
Comment (harshith-venkatesh) [inline components/ThemeToggler.vue:10]: NIT: What is your opinion of keeping `setCookie` in a utility function

File: components/ThemeToggler.vue [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+      : (this.isDarkTheme = false)
+  },
+  methods: {
+    toggleClicked(value) {
+      this.isClicked = !value
+    },
+    toggleTheme() {
+      this.isDarkTheme = !this.isDarkTheme
+      this.isDarkTheme
+        ? (this.themeImage = 'sun.png')
```
Comment (harshith-venkatesh) [inline components/ThemeToggler.vue:46]: I am observing repetition of `themeImage` and `document.documentElement.className` could we look into way to reduce this

File: components/ThemeToggler.vue [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        : (this.themeImage = 'moon.png')
+      this.isDarkTheme
+        ? (document.documentElement.className = 'dark')
+        : (document.documentElement.className = 'light')
+      const currentTheme = this.isDarkTheme ? 'dark' : 'light'
```
Comment (harshith-venkatesh) [inline components/ThemeToggler.vue:51]: ref

File: pages/index.vue [hunk 6]
```diff
@@ -15,6 +17,17 @@ import Clock from '../components/Clock.vue'
 import Days from '../components/Days.vue'
 import Goals from '../components/Goals.vue'
 import Footer from '../components/Footer.vue'
+
+function getCookie(name) {
+  const nameEQ = name + '='
```
Comment (harshith-venkatesh) [inline pages/index.vue:22]: What is the significance of EQ here

File: pages/index.vue [hunk 7]
```diff
@@ -23,6 +36,23 @@ export default {
...
+    return {
+      theme: '',
+    }
+  },
+  beforeMount() {
+    const currentTheme = getCookie('theme')
+    if (currentTheme) {
+      this.theme = currentTheme
+      this.theme === 'dark'
+        ? (document.documentElement.className = 'dark')
```
Comment (harshith-venkatesh) [inline pages/index.vue:49]: Is there any problem with this way?
`document.documentElement.className = this.theme === 'dark' ? 'dark':'light'`

**Review by harshith-venkatesh — COMMENTED** [review]
Great work Pavan

File: components/ThemeToggler.vue [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+    </span>
+  </div>
+</template>
+
+<script>
+function setCookie(name, value, days) {
+  let expires = ''
+  if (days) {
+    const date = new Date()
+    date.setTime(date.getTime() + days * 24 * 60 * 60 * 1000)
```
Comment (harshith-venkatesh) [inline components/ThemeToggler.vue:14]: Just for the information: How are we getting days here 
Is it not usually 28-31 days

File: components/ThemeToggler.vue (hunk 3 above)
Comment (Pavangbhat) [reply components/ThemeToggler.vue:10]: Right :100:

File: pages/index.vue (hunk 6 above)
Comment (Pavangbhat) [reply pages/index.vue:22]: Since it's not just a name anymore it has **'='** making it a **name equation**

File: pages/index.vue (hunk 7 above)
Comment (Pavangbhat) [reply pages/index.vue:49]: Right should have made it simple

File: components/ThemeToggler.vue (hunk 8 above)
Comment (Pavangbhat) [reply components/ThemeToggler.vue:14]: Yes it is, but as per the discussion we decided to go for 30 days expiry

File: helpers/cookies.js [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+}
+
+export function setCookie(name, value, days) {
+  let expires = ''
+  if (days) {
+    const date = new Date()
+    date.setTime(date.getTime() + days * 24 * 60 * 60 * 1000)
+    expires = '; expires=' + date.toUTCString()
+  }
+  document.cookie = name + '=' + (value || '') + expires + '; path=/'
```
Comment (akshay1502) [inline helpers/cookies.js:19]: Could we use template strings here? Also, add `domain` value too here.

**Comment by RohitKuwar** [comment]
@Pavangbhat Can we please fix the dark-mode icon place for the mobile devices?

Screenshot (986)

**Comment by RohitKuwar** [comment]
@Pavangbhat If possible, can we please add a little transition while the theme is changing?

**Comment by RohitKuwar** [comment]
@Pavangbhat Can we please place the dark-theme icon at the extreme right side of the navbar?

Screenshot (987)

like this:
Screenshot (988)

**Comment by Pavangbhat** [comment]
> @Pavangbhat If possible, can we please add a little transition while the theme is changing?

Sure :100:

**Review by Pavangbhat — COMMENTED** [review]
- [x] responsive design
- [x] transition effect on theme change
https://user-images.githubusercontent.com/45538392/141066513-82cc5c8e-b64a-4368-ba13-e7d827fa62e9.mp4

**Comment by RohitKuwar** [comment]
Great work @Pavangbhat!!!

**Review by RohitKuwar — COMMENTED** [review]
Great work @Pavangbhat !!!

File: helpers/document.js [hunk 10]
```diff
@@ -0,0 +1,3 @@
+export function addClassToDocument(className) {
+  document.documentElement.className = className
+}
```
[the remark this answers is not in the document]
Comment (Pavangbhat) [reply helpers/document.js:3]: I was using **document.documentElement.className** in two files so thought of creating a function that can be used.
But I think it's not going to be much useful, so let me know if I should revert it

**Pull request closed without merging**